server/src/main/java/com/ncc/neon/better/NlpModule.java

```diff
@@ -114,4 +147,30 @@ protected Disposable handleNlpOperationError(WebClientResponseException err, Str
                 .contentType(MediaType.APPLICATION_JSON)
                 .body(BodyInserters.fromValue(data));
     }
+
+    private WebClient buildNlpWebClient(String name) {
+        String url = "http://";
+        String host = System.getenv().getOrDefault(name.toUpperCase() + "_HOST", "localhost");
+        String port = env.getProperty(name + ".port");
```

Comment:
Seems weird to use completely different rules for host and port env vars. For example, FOO_HOST and foo.port. Why not FOO_PORT?

Reply:
This is because the host and port values are coming from different sources.  Host is coming from an environment variable, where we are defining the naming convention to separate words with `_` and all uppercase letters.  Port is coming from `application.properties`, where it's Spring naming convention to separate words with `.` and all lowercase letters.

Reply:
Ah, I didn't notice that. But that raises the question: Why are host and port coming from two different source? Shouldn't they be defined the same way/in the same place?

Reply:
Discussed over phone.